Sources/XcodeGenKit/SettingsBuilder.swift

```diff
@@ -24,18 +24,47 @@ extension ProjectSpec {
         }
 
         buildSettings += getBuildSettings(settings: settings, config: config)
+        if let configPath = configFiles[config.name] {
+            // Do not overwrite project xcconfig's value.
+            if let configFile = try? XCConfig(path: basePath + configPath) {
+                for (k, _) in configFile.flattenedBuildSettings() {
+                    // FIXME: Catch platform specifier. e.g. LD_RUNPATH_SEARCH_PATHS[sdk=iphone*]
+                    buildSettings.removeValue(forKey: k)
+                    buildSettings.removeValue(forKey: "\"\(k)\"")
+                }
+            }
+        }
 
         return buildSettings
     }
 
-    public func getTargetBuildSettings(target: Target, config: Config) -> BuildSettings {
+    public func getTargetBuildSettings(basePath: Path, target: Target, config: Config) -> BuildSettings {
         var buildSettings = BuildSettings()
 
         buildSettings += SettingsPresetFile.platform(target.platform).getBuildSettings()
         buildSettings += SettingsPresetFile.product(target.type).getBuildSettings()
         buildSettings += SettingsPresetFile.productPlatform(target.type, target.platform).getBuildSettings()
         buildSettings += getBuildSettings(settings: target.settings, config: config)
```

Comment:
I think the clearing should happen before this line, so custom build settings don't get removed, just the setting presets